### PR TITLE
Adding Support for Apple Music

### PR DESCRIPTION
**Pull request by 4-Eyes**
# Apple Music Scrobbling
Adds a notification listener service for parsing notifications put out by Apple Music for the the purposes of identifying songs being played by the application. Appropriate broadcasts are then sent (as if Apple Music were doing it) for Simple Last.fm Scrobbler to process and scrobble. This service can be enabled within settings.

It should be noted that this method of detected songs is not 100% accurate as there are times when notifications from Apple Music can disappear (quite often when browsing the application). However, this appears to be the method that other applications (such as Musixmatch) use to find out what Apple Music is playing.

Solves Issue #232

### CODE REVIEW

File: app/src/main/java/com/adam/aslfms/service/applemusic/TrackData.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                return overlapTime;
+            }
+            playTimes.add(lastPlayTime);
+        }
+        return 0;
+    }
+
+    boolean sameTrack(TrackData other) {
+        return this.title != null && this.artist != null && this.album != null &&
+                other.title != null && other.artist != null && other.album != null &&
```
Comment (fallaciousreasoning) [inline app/src/main/java/com/adam/aslfms/service/applemusic/TrackData.java]: Do you need to check whether the fields on other aren't null? "some_string".equals(null) should be fine, right?

Comment (fallaciousreasoning) [inline app/src/main/java/com/adam/aslfms/service/applemusic/TrackData.java]: Also, would a check that other != null be worth adding?

Comment (fallaciousreasoning) [inline app/src/main/java/com/adam/aslfms/service/applemusic/TrackData.java]: Watch me nitpick :P

Comment (4-Eyes) [inline app/src/main/java/com/adam/aslfms/service/applemusic/TrackData.java]: Are you happy now?

Comment (fallaciousreasoning) [inline app/src/main/java/com/adam/aslfms/service/applemusic/TrackData.java]: No, but that's unrelated :P (LGTM)

**Comment by HumbleBeeBumbleBee** [comment]
I will test as soon as I can. Thank you for your efforts. :-)

**Pull request merged**